### PR TITLE
Fixed scrolling on filter screen

### PR DESCRIPTION
**Pull request by nschwerd**
Fixed veteran check box behavior
Moved qualifying classes from model to new util pkg

### CODE REVIEW

File: app/src/main/res/layout/activity_filter_shelters.xml [hunk 1]
```diff
@@ -1,135 +1,143 @@
 <?xml version="1.0" encoding="utf-8"?>
-<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<android.support.v4.widget.NestedScrollView xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (jmschonfeld) [inline app/src/main/res/layout/activity_filter_shelters.xml]: Is there a reason why this is a `NestedScrollView` and not just a standard `ScrollView`?

File: app/src/main/res/layout/activity_filter_shelters.xml [hunk 2]
```diff
@@ -1,135 +1,143 @@
...
     xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
+    android:fillViewport="true"
     android:orientation="vertical"
-    tools:context="edu.gatech.spacebarz.buzzshelter.FilterSheltersActivity"
```
Comment (jmschonfeld) [inline app/src/main/res/layout/activity_filter_shelters.xml]: Is there a reason this line was deleted?

Comment (nschwerd) [reply app/src/main/res/layout/activity_filter_shelters.xml]: It is there on new line 14

File: app/src/main/res/layout/activity_filter_shelters.xml (hunk 1 above)
Comment (nschwerd) [reply app/src/main/res/layout/activity_filter_shelters.xml]: Didn't show up in autocomplete.
Can change if functionality is preserved.

Comment (jmschonfeld) [reply app/src/main/res/layout/activity_filter_shelters.xml]: It shows up for me in autocomplete (the tag is just `<ScrollView>`). Nested scroll view is probably fine but is meant for when its a scroll view inside of a scroll view, which causes the issues we're having on the shelter list so may be better to use a simple `ScrollView`

Comment (nschwerd) [reply app/src/main/res/layout/activity_filter_shelters.xml]: Strange, maybe I need to have intellij rebuild its autocomplete dict.
I'm all for less problems, let's make the change.

**Comment by nschwerd** [comment]
Just changed to ScrollView and added some interaction behaviors. LMK what you think @jmschonfeld

**Pull request merged**